doc/error-catalog.md

```diff
@@ -174,6 +174,35 @@ Examples:
       echo yes
     }
 
+### OILS-ERR-15
+
+Incorrect:
+
+    # Expression mode
+    if (a || b) {
```

Comment:
maybe add `if (a || b && c)`

and contrast with `if (a or b and c)`

to show both operators

I guess we can also show negation

```
if (!a || b && c)

if (not a or b and c)

```